Plugins/GeoModel/src/GeoModelToDetectorVolume.cpp

```diff
@@ -96,93 +57,94 @@ std::shared_ptr<Experimental::DetectorVolume> convertVolume(
-            std::make_shared<TrapezoidVolumeBounds>(y1, y2, z, x1);
+        bounds = std::make_shared<TrapezoidVolumeBounds>(y1, y2, z, x1);
         auto rotationAngle = M_PI / 2;
-        GeoTrf::Transform3D newTransform = transform *
-                                           GeoTrf::RotateZ3D(rotationAngle) *
-                                           GeoTrf::RotateX3D(rotationAngle);
-        return Experimental::DetectorVolumeFactory::construct(
-            portalGenerator, context, name, newTransform, bounds, sensSurfaces,
-            a, Experimental::tryNoVolumes(),
-            Experimental::tryAllPortalsAndSurfaces());
+        newTrf = trf * GeoTrf::RotateZ3D(rotationAngle) *
+                 GeoTrf::RotateX3D(rotationAngle);
       } else {
-        std::shared_ptr<TrapezoidVolumeBounds> bounds =
-            std::make_shared<TrapezoidVolumeBounds>(y2, y1, z, x1);
+        bounds = std::make_shared<TrapezoidVolumeBounds>(y2, y1, z, x1);
         auto rotationAngle = M_PI;
-        GeoTrf::Transform3D newTransform =
-            transform * GeoTrf::RotateX3D(rotationAngle) *
-            GeoTrf::RotateZ3D(rotationAngle / 2) *
-            GeoTrf::RotateX3D(rotationAngle / 2);
-        return Experimental::DetectorVolumeFactory::construct(
-            portalGenerator, context, name, newTransform, bounds, sensSurfaces,
-            a, Experimental::tryNoVolumes(),
-            Experimental::tryAllPortalsAndSurfaces());
+        newTrf = trf * GeoTrf::RotateX3D(rotationAngle) *
+                 GeoTrf::RotateZ3D(rotationAngle / 2) *
+                 GeoTrf::RotateX3D(rotationAngle / 2);
       }
     } else {
       throw std::runtime_error("FATAL: Translating GeoTrd to ACTS failed");
     }
-  } else if (shape.typeID() == GeoShapeUnion::getClassTypeID()) {
+  }
+
+  else if (shape.typeID() == GeoShapeUnion::getClassTypeID()) {
     const GeoShapeUnion* unionShape =
         dynamic_cast<const GeoShapeUnion*>(&shape);
     double xmin{0}, xmax{0}, ymin{0}, ymax{0}, zmin{0}, zmax{0};
     unionShape->extent(xmin, ymin, zmin, xmax, ymax, zmax);
-    std::shared_ptr<CuboidVolumeBounds> bounds =
-        std::make_shared<CuboidVolumeBounds>(
-            (xmax - xmin) / 2, (ymax - ymin) / 2, (zmax - zmin) / 2);
-    return Experimental::DetectorVolumeFactory::construct(
-        portalGenerator, context, name, transform, bounds,
-        Experimental::tryAllPortalsAndSurfaces());
+    bounds = std::make_shared<CuboidVolumeBounds>(
+        (xmax - xmin) / 2, (ymax - ymin) / 2, (zmax - zmin) / 2);
   } else if (shape.typeID() == GeoShapeSubtraction::getClassTypeID()) {
     // Go down the left side (opA) of the subtraction until we reach a normal
     // shape
     const GeoShapeSubtraction* subtractionShape =
         dynamic_cast<const GeoShapeSubtraction*>(&shape);
     const GeoShape* shapeA = subtractionShape->getOpA();
-    return convertVolume(context, *shapeA, name, transform, sensitives);
+    return package(trf, *shapeA);
```

Comment:
is this practice really converts the subtraction shape? is it tested? Maybe I am missing something, but I have the impression it just converts the first shape of the subtraction that it finds and falls in any of these shape cases.

Reply:
I am not sure, but I don't see in what way this is logically different from the version before my changes. This is not tested though so it is hard to tell. What would be a good way of testing?

Reply:
Having done the analogous conversion for the surfaces I am fairly confident that this logic is appropriate